### PR TITLE
Trade Widget

### PR DESCRIPTION
**Pull request by pingustar**
#236 #292 #84 #378

### CODE REVIEW

**Review by RanCohenn — CHANGES_REQUESTED** [review]
Maybe we should add limit into the widget logic, with certain separation.

Before this is merged into main we need to:

- [x] Remove the beta url
- [x] remove old swap code

File: src/elements/trade/TradeWidgetCTA.tsx
Comment (pingustar) [reply src/elements/trade/TradeWidgetCTA.tsx:42]: thanks, will fix

**Comment by pingustar** [comment]
> Maybe we should add limit into the widget logic, with certain separation.

lets tackle limit in a separate issue and pr.

**Comment by ashachaf** [comment]
- [ ] 
image
flow:
- set trade from BNT to ETH
- enter bnt amount
- wait 1-2 seconds (before the rate appears)
- change token to DAI

expected behavior should:
- keep the input BNT amount 
- re-calculate the rate based on the input amount and destination token

**Comment by ashachaf** [comment]
- [ ] handle wETH
image
reference:
https://github.com/bancorprotocol/webapp-v2/issues/210
https://github.com/bancorprotocol/webapp-v2/issues/64

**Comment by ashachaf** [comment]
- [ ] make sure when selecting weth in the from, we automatically select eth as the destination token (lock it from changes)
- [ ] when clicking the switch option, set eth as the from token and reset the destination token to show "select token"

image
image

**Review by zavelevsky — APPROVED** [review]
LGTM

**Comment by ashachaf** [comment]
- [ ] close "approve" popup should reset the "trade" CTA
image
flow:
1. set trade from a token that requires approve
2. when the approve popup shows up, close it
expected: trade CTA to be enabled
actual: it is disabled and force to refresh

- [ ] input field shows NaN
image
flow:
1. set trade from a token that requires approve (only happens when view is on USD)
2. when the approve popup shows up, close it
3. change token to DAI
4. try to change input value

- [ ] select full balance on USD view input incorect values
image
flow:
1. set the trade to USD view
2. select all (click the balance)
expected: see the full balance in the from token field
actual: we pull the token amount as the USD input.

**Review by RanCohenn — DISMISSED** [review]
LGTM

**Review by ashachaf — DISMISSED** [review]

**Comment by ashachaf** [comment]
- [ ] input amount edge case
flow:
1. set input to usd
2. set $333 DAI
3. change token from DAI to ETH
expected: $333 worth of ETH
actual: the DAI token amount is used as ETH amount

File: src/elements/trade/TradeWidgetCTA.tsx
Comment (zavelevsky) [inline src/elements/trade/TradeWidgetCTA.tsx:36]: I'd wrap with useMemo

File: src/elements/trade/TradeWidgetInput.tsx
Comment (zavelevsky) [inline src/elements/trade/TradeWidgetInput.tsx:39]: are you trying to achieve something like this? https://reactjs.org/docs/hooks-reference.html#useimperativehandle

**Pull request merged**
